cmds/core/xargs/xargs.go

```diff
@@ -16,17 +16,38 @@ import (
 )
 
 const defaultMaxArgs = 5000
+const tty = "/dev/tty"
 
-func main() {
-	var maxNumber = flag.Int("n", defaultMaxArgs, "max number of arguments per command")
+type params struct {
+	tty     string
+	maxArgs int
+	trace   bool
+	promt   bool
+}
+
+func parseParams() params {
+	var maxArgs = flag.Int("n", defaultMaxArgs, "max number of arguments per command")
 	var trace = flag.Bool("t", false, "enable trace mode, each command is written to stderr")
+	var promt = flag.Bool("p", false, "the user is asked whether to execute utility at each invocation")
+
 	flag.Parse()
-	if err := run(os.Stdin, os.Stdout, os.Stderr, *maxNumber, *trace, flag.Args()...); err != nil {
+	p := params{
+		maxArgs: *maxArgs,
+		trace:   *trace || *promt,
+		promt:   *promt,
+		tty:     tty,
+	}
+
+	return p
+}
+
+func main() {
+	if err := run(os.Stdin, os.Stdout, os.Stderr, parseParams(), flag.Args()...); err != nil {
 		log.Fatal(err)
 	}
 }
 
-func run(stdin io.Reader, stdout, stderr io.Writer, maxArgs int, trace bool, args ...string) error {
+func run(stdin io.Reader, stdout, stderr io.Writer, p params, args ...string) error {
```

Comment:
should it be 
``*params? ```
might it be 
```
func (c*cmd) run(args...string)
```
similarly to others? i.e. follow the exec.Cmd pattern?
Also s/promt/prompt/ :-)

Reply:
updated to command pattern and fixed spelling :)